tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -362,6 +366,7 @@ class ASTBuildOperation
   /// Transition the build operation to \p NewState, asserting that the current
   /// state is \p ExpectedOldState.
   void transitionToState(State NewState, State ExpectedOldState) {
+    llvm::sys::ScopedLock L(OperationStateMtx);
```

Comment:
It seems like there could still be a race to me here. This fixes two threads changing/reading at the same time, but it won't fix the ordering. I imagine that could mean we just assert now 🤔?